File: gisPhpApi/phpLib/class_pop_junction.php
<?php
class class_pop_junction{
	
	function get_junction_by_name($name_loc){
                
		$junction_info=array();
		
		$name_loc = preg_replace('/\s+/', '%20', $name_loc);
		//echo $name_loc;
		//$url="http://52.74.144.159:8080/fulltext/fulltextsearch?q=$name_loc&placetype=road&format=JSON&__checkbox_indent=true&from=1&to=1";
                $url="http://52.74.144.159:8080/fulltext/fulltextsearch?q=$name_loc&format=JSON&__checkbox_indent=true&from=1&to=1";
		$json = file_get_contents($url);
		$data = json_decode($json, TRUE);
		//print_r($data);

		$feature_id=$data['response']['docs'][0]['feature_id'];
		$name=$data['response']['docs'][0]['name'];
		$lat=$data['response']['docs'][0]['lat'];
		$lng=$data['response']['docs'][0]['lng'];
		//$feature_code=$data['response']['docs'][0]['feature_code'];
		//$placetype=$data['response']['docs'][0]['placetype'];//ROAD
		//echo "CHURAHA INFO, FeatureID=".$feature_id.":Name=".$name.":Lat=".$lat.":Lng=".$lng.":FeatureCode=".$feature_code;
		$location_name=$name;		
		$junction_info[]=array('code'=>$feature_id,'location'=>$location_name,'lat'=>$lat,'lng'=>$lng);
		
		return $junction_info;
	}
	function  get_junction_by_latlng($lat,$lng)
	{
		
		$junction_info=array();
		$url="http://52.74.144.159:8080/geoloc/geolocsearch?lat=$lat&lng=$lng&placetype=road&format=JSON&distance=true&from=1&to=1";
		//echo $url;
                $json = file_get_contents($url);
		$data = json_decode($json, TRUE);
		//print_r($data);
		
		$feature_id=$data['result'][0]['featureId'];
		$name=$data['result'][0]['name'];
		$lat=$data['result'][0]['lat'];
		$lng=$data['result'][0]['lng'];
		$feature_code=$data['result'][0]['featureCode'];
		$distance=$data['result'][0]['distance'];
		//$placetype=$data['result'][0]['placetype'];//ROAD
		//echo "CHURAHA INFO, FeatureID=".$feature_id.":Name=".$name.":Lat=".$lat.":Lng=".$lng.":FeatureCode=".$feature_code;
		$location_name=$name;		
		
		$junction_info[]=array('code'=>$feature_id,'location'=>$name,'lat'=>$lat,'lng'=>$lng,'distance'=>$distance);
		return $junction_info;
	}
	
	function  get_junction_near_by_latlng($lat,$lng,$radius)
	{
		
		$junction_info=array();
		$url="http://52.74.144.159:8080/geoloc/geolocsearch?lat=$lat&lng=$lng&placetype=road&format=JSON&distance=true&radius=$radius&from=1&to=1";
		$json = file_get_contents($url);
		$data = json_decode($json, TRUE);
		//print_r($data);
		
		$feature_id=$data['result'][0]['featureId'];
		$name=$data['result'][0]['name'];
		$lat=$data['result'][0]['lat'];
		$lng=$data['result'][0]['lng'];
		$feature_code=$data['result'][0]['featureCode'];
		$distance=$data['result'][0]['distance'];
		//$placetype=$data['result'][0]['placetype'];//ROAD
		//echo "CHURAHA INFO, FeatureID=".$feature_id.":Name=".$name.":Lat=".$lat.":Lng=".$lng.":FeatureCode=".$feature_code;
		$location_name=$name;		
		
		$junction_info[]=array('code'=>$feature_id,'location'=>$name,'lat'=>$lat,'lng'=>$lng,'distance'=>$distance);
		return $junction_info;
	}
	
	function get_junction_by_code($code)
	{
		//============This will return from pOSTGRES sql=============================
		include('gis_connection.php');
		if(!$db_connection){
		  echo "Error : Unable to open database\n";
		} else {
		  //echo "Opened database successfully\n";
		}
		$junction_info=array();
		//$code="100220619";
		$query="SELECT name,astext(location) as lnglat FROM road where featureid=$code ";
		//echo $query;
		$result1 = pg_query($db_connection, $query);		
		$row1=pg_fetch_object($result1);
		$name=$row1->name;
		$lnglat=explode(" ",$row1->lnglat);
		$lng_tmp=explode("(",$lnglat[0]);
		$lat_tmp=explode(")",$lnglat[1]);
		//$featurecode=$row1->featurecode;
		$location_name=$name;
		//return "Name=".$name.":Lng=".$lng_tmp[1].":Lat=".$lat_tmp[0];
		$junction_info[]=array('code'=>$code,'location'=>$name,'lat'=>$lat_tmp[0],'lng'=>$lng_tmp[1]);
		return $junction_info;
	}
	
	
        function get_junction_by_latlng_array($lat_lng_array,$radius)
	{
		//============This will return from pOSTGRES sql=============================
		include('gis_connection.php');
		if(!$db_connection){
		  echo "Error : Unable to open database\n";
		} else {
		 // echo "Opened database successfully\n";
		}
		$junction_info=array();
                //echo count($lat_lng_array);
		foreach($lat_lng_array as $lat_lng)
                {
                    $lat=$lat_lng['lat'];
                    $lng=$lat_lng['lng'];							
                    $lat_minus= $lat-1;
                    $lat_plus= $lat+1;
                    $lng_minus= $lng-1;
                    $lng_plus= $lng+1;
                    
                    $query="SELECT id, name, astext(location) as lnglat,featureid, CAST (st_distance_sphere(location, st_setsrid(st_makepoint($lng,$lat),4326)) AS INT) AS d FROM road WHERE location && 'BOX3D($lng_minus $lat_minus,$lng_plus $lat_plus)'::box3d and name!='' ORDER BY location <-> st_setsrid(st_makepoint($lng ,$lat), 4326)  LIMIT 1";
                    //echo $query;
                    $result1 = pg_query($db_connection, $query);		
                    $row1=pg_fetch_object($result1);
                    $junction_name=$row1->name;
                    $junction_code=$row1->featureid;
                    $get_radius=$row1->d;
                    if($get_radius <= $radius)
                    {
                        $junction_info[]=array('lat'=>$lat,'lng'=>$lng,'junction_name'=>$junction_name,'junction_code'=>$junction_code,'get_radius'=>$get_radius,'landmark'=>'-');
                    }
                    else
                    {
                        if($get_radius < 5000)
                        {
                            $junction_info[]=array('lat'=>$lat,'lng'=>$lng,'junction_name'=>$junction_name,'junction_code'=>$junction_code,'get_radius'=>$get_radius,'landmark'=>'-');
                        }
                        else
                        {
                            $junction_info[]=array('lat'=>$lat,'lng'=>$lng,'junction_name'=>'-','junction_code'=>'-','get_radius'=>'-','landmark'=>'-');
                        }
                    }
                }               
               
		return $junction_info;
	}
	
}
?>

File: gisJavaApi/src/com/iespl/gisgraphy/class_pop_junction.java
package com.iespl.gisgraphy;
import java.util.ArrayList;
import java.util.regex.*;
import java.sql.*;

public class class_pop_junction {
	String lat;
	String lng;
	String code;
	String churaha_name="";
	String churaha_code="";
	
	private ArrayList<LatLng >  latlngData = new ArrayList<LatLng>();
	
	public ArrayList<LatLng> getLatlngData() {
		return latlngData;
	}
	
	gis_connection gis_con = new gis_connection();
	//========pop by lat lng
	public class_pop_junction(String lt, String lg){
		lat=lt;
		lng=lg;
		//System.out.println("Latitude:"+lat);
		//System.out.println("Longitude:"+lng);		
		/*try{
			Class.forName("org.postgresql.Driver");
		}catch(ClassNotFoundException e){
			//System.out.println("Where is your PostgreSQL JDBC Driver? Included in your library path!");
			e.printStackTrace();
			return;
		}
		//System.out.println("PostgreSQL JDBC Driver Registered!");
		Connection connection=null;
		try{
			//connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5432/gisgraphy","postgres","neon04$");
			connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5433/gisgraphy","postgres","neon04$");
			//connection = DriverManager.getConnection("jdbc:postgresql://52.74.144.159:5432/gisgraphy","postgres","neon04$");
		}catch(SQLException e){
			//System.out.println("Connection Failed!Check output console");
			e.printStackTrace();
			return;
		}*/
		if(gis_con.connection !=null){
			//System.out.println("You made it, take control your database now!");
			//Create a Statement class to execute the SQL statement
		    try {
				Statement stmt = gis_con.connection.createStatement();
				 //Execute the SQL statement and get the results in a Resultset
				String query="SELECT id, name, astext(location) as lnglat,featureid, CAST (st_distance_sphere(location, st_setsrid(st_makepoint("+lng+","+lat+"),4326)) AS INT) AS d FROM road ORDER BY location <-> st_setsrid(st_makepoint("+lng+","+lat+"), 4326)  LIMIT 1";
			    ResultSet rs1 = stmt.executeQuery(query);
			    // Iterate through the ResultSet, displaying two values
			    // for each row using the getString method
			 
			    while (rs1.next()){
			        //System.out.println("Name= " + rs.getString("name") + " Code= " + rs.getString("featureId"));
			    	churaha_name=rs1.getString("name");
			    	churaha_code=rs1.getString("featureId");
			    }
			} catch (SQLException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			
		}
		else{
			System.out.print("Failed to make connection");
		}
	}
	//========pop by latlng array
	public class_pop_junction(ArrayList<LatLng> latlngArray,int radius){
			
		try{
			Class.forName("org.postgresql.Driver");
		}catch(ClassNotFoundException e){
			//System.out.println("Where is your PostgreSQL JDBC Driver? Included in your library path!");
			e.printStackTrace();
			return;
		}
		//System.out.println("PostgreSQL JDBC Driver Registered!");
		Connection connection=null;
		try{
			//connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5432/gisgraphy","postgres","neon04$");
			connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5433/gisgraphy","postgres","neon04$");
			//connection = DriverManager.getConnection("jdbc:postgresql://52.74.144.159:5432/gisgraphy","postgres","neon04$");
		}catch(SQLException e){
			//System.out.println("Connection Failed!Check output console");
			e.printStackTrace();
			return;
		}
		if(connection !=null){
			//System.out.println("You made it, take control your database now!");
			//Create a Statement class to execute the SQL statement
		    try {
				Statement stmt = connection.createStatement();
				
				
				for(LatLng data : latlngArray){
					//System.out.println("data : "+data.getLat());
					//System.out.println("data : "+data.getLng());
					lat=data.getLat();
					lng=data.getLng();								
			
					double lat_minus= Double.parseDouble(lat)-1;
					double lat_plus= Double.parseDouble(lat)+1;
					double lng_minus= Double.parseDouble(lng)-1;
					double lng_plus= Double.parseDouble(lng)+1;
					//Execute the SQL statement and get the results in a Resultset
					String query="SELECT id, name, astext(location) as lnglat,featureId, CAST (st_distance_sphere(location, st_setsrid(st_makepoint("+lng+","+lat+"),4326)) AS INT) AS d FROM road WHERE location && 'BOX3D("+lng_minus+" "+lat_minus+","+lng_plus+" "+lat_plus+")'::box3d and name!='' ORDER BY location <-> st_setsrid(st_makepoint("+lng+","+lat+"), 4326)  LIMIT 1";
					//String query="SELECT id, name, astext(location) as lnglat,featureId, CAST (st_distance_sphere(location, st_setsrid(st_makepoint("+lng+","+lat+"),4326)) AS INT) AS d FROM road WHERE name!='' ORDER BY location <-> st_setsrid(st_makepoint("+lng+","+lat+"), 4326)  LIMIT 1";
					//String query="SELECT id, name, astext(location) as lnglat,featureId FROM road ORDER BY location <-> st_setsrid(st_makepoint("+lng+","+lat+"), 4326)  LIMIT 1";
					//System.out.println(query);
					//long starttime  = System.currentTimeMillis();
					long starttime  =System.nanoTime();
					ResultSet rs1 = stmt.executeQuery(query);
					//long endtime  = System.currentTimeMillis();
					long endtime  = System.nanoTime();
					//System.out.println("start time latlng to get"+(endtime-starttime));
					// Iterate through the ResultSet, displaying two values
					// for each row using the getString method
					String get_radius="";
					while (rs1.next()){
						//System.out.println("Name= " + rs.getString("name") + " Code= " + rs.getString("featureId"));
						churaha_name=rs1.getString("name");
						churaha_code=rs1.getString("featureId");
						get_radius=rs1.getString("d");
						//System.out.println("radius=:"+get_radius);
						//System.out.println("end time latlng to get"+System.currentTimeMillis());
						if(Double.parseDouble(get_radius) <= radius)
						{
							LatLng item = new LatLng(lat, lng, churaha_name, churaha_code,get_radius,"-");
							latlngData.add(item);
						}
						else
						{
							if(Double.parseDouble(get_radius) < 5000)
							{
								LatLng item = new LatLng(lat, lng, churaha_name, churaha_code,get_radius,"-");
								latlngData.add(item);
							}
							LatLng item = new LatLng(lat, lng, "-", "-","-","-");
							latlngData.add(item);
						}
						
						
					}
				}
			} catch (SQLException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			
		}
		else{
			System.out.print("Failed to make connection");
		}
	}
	//===================pop by by code===========
	
	public class_pop_junction(String cde){
		code=cde;
		/*try{
			Class.forName("org.postgresql.Driver");
		}catch(ClassNotFoundException e){
			//System.out.println("Where is your PostgreSQL JDBC Driver? Included in your library path!");
			e.printStackTrace();
			return;
		}
		//System.out.println("PostgreSQL JDBC Driver Registered!");
		Connection connection=null;
		try{
			//connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5432/gisgraphy","postgres","neon04$");
			connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5433/gisgraphy","postgres","neon04$");
			//connection = DriverManager.getConnection("jdbc:postgresql://52.74.144.159:5432/gisgraphy","postgres","neon04$");
		}catch(SQLException e){
			//System.out.println("Connection Failed!Check output console");
			e.printStackTrace();
			return;
		}*/
		if(gis_con.connection !=null){
			//System.out.println("You made it, take control your database now!");
			//Create a Statement class to execute the SQL statement
		    try {
				Statement stmt = gis_con.connection.createStatement();
				 //Execute the SQL statement and get the results in a Resultset
				String query="SELECT name,astext(location) as lnglat,featureId FROM road where featureid="+code;
			    ResultSet rs1 = stmt.executeQuery(query);
			    // Iterate through the ResultSet, displaying two values
			    // for each row using the getString method
			 
			    while (rs1.next()){
			        //System.out.println("Name= " + rs.getString("name") + " Code= " + rs.getString("featureId"));
			    	churaha_name=rs1.getString("name");
			    	churaha_code=rs1.getString("featureId");
			    	String lnglat="";
			    	lnglat=rs1.getString("lnglat");
			    	//System.out.println(lnglat);
			    
			    	
			    	// String to split. 			    	
			    	String[] temp1;
			    	// delimiter 
			    	String delimiter1 = " ";
			    	String delimiter2 = "("; 
			    	String delimiter3 = ")";
			    	// given string will be split by the argument delimiter provided. 
			    	temp1 = lnglat.split(delimiter1);
			    	// print substrings 
			    	
			    	String[] lng1;String[] lat1;
			    	//System.out.println(temp1[0]+" & "+temp1[1]);			    
			    	String lng2=temp1[0];
			    	lng1=lng2.split("\\(");  //(lng at 1
			    	lng=lng1[1];
			    	String lat2=temp1[1];
			    	lat1=lat2.split("\\)");  //(lat 0
			    	lat=lat1[0];
			    
			    }
			} catch (SQLException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			
		}
		else{
			System.out.print("Failed to make connection");
		}
		
	}
	
	//========pop by code array
	public class_pop_junction(ArrayList<LatLng> codeArray){
			
		/*try{
			Class.forName("org.postgresql.Driver");
		}catch(ClassNotFoundException e){
			//System.out.println("Where is your PostgreSQL JDBC Driver? Included in your library path!");
			e.printStackTrace();
			return;
		}
		//System.out.println("PostgreSQL JDBC Driver Registered!");
		Connection connection=null;
		try{
			//connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5432/gisgraphy","postgres","neon04$");
			connection = DriverManager.getConnection("jdbc:postgresql://127.0.0.1:5433/gisgraphy","postgres","neon04$");
			//connection = DriverManager.getConnection("jdbc:postgresql://52.74.144.159:5432/gisgraphy","postgres","neon04$");
		}catch(SQLException e){
			//System.out.println("Connection Failed!Check output console");
			e.printStackTrace();
			return;
		}*/
		if(gis_con.connection !=null){
			//System.out.println("You made it, take control your database now!");
			//Create a Statement class to execute the SQL statement
		    try {
				Statement stmt = gis_con.connection.createStatement();
				
				
				for(LatLng data : codeArray){
					//System.out.println("data : "+data.getLat());
					//System.out.println("data : "+data.getLng());
					code=data.getLocationCode();
												
			
					//Execute the SQL statement and get the results in a Resultset
					
					String query="SELECT name,astext(location) as lnglat,featureId FROM road where featureid="+code;
					//System.out.println(query);
					ResultSet rs1 = stmt.executeQuery(query);
					// Iterate through the ResultSet, displaying two values
					// for each row using the getString method
			 
					while (rs1.next()){
						
						churaha_name=rs1.getString("name");
				    	churaha_code=rs1.getString("featureId");
				    	String lnglat="";
				    	lnglat=rs1.getString("lnglat");
				    	//System.out.println(lnglat);
				    
				    	
				    	// String to split. 			    	
				    	String[] temp1;
				    	// delimiter 
				    	String delimiter1 = " ";
				    	String delimiter2 = "("; 
				    	String delimiter3 = ")";
				    	// given string will be split by the argument delimiter provided. 
				    	temp1 = lnglat.split(delimiter1);
				    	// print substrings 
				    	
				    	String[] lng1;String[] lat1;
				    	//System.out.println(temp1[0]+" & "+temp1[1]);			    
				    	String lng2=temp1[0];
				    	lng1=lng2.split("\\(");  //(lng at 1
				    	lng=lng1[1];
				    	String lat2=temp1[1];
				    	lat1=lat2.split("\\)");  //(lat 0
				    	lat=lat1[0];
						
						LatLng item = new LatLng(lat, lng, churaha_name, churaha_code,"-","-");
						latlngData.add(item);
					}
				}
			} catch (SQLException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			
		}
		else{
			System.out.print("Failed to make connection");
		}
	}
	//===================end pop by by code===========
	
	public ArrayList<String> Data(){
		ArrayList<String> values = new ArrayList<String>();
		String final_data=String.valueOf(lat)+":"+String.valueOf(lng)+":"+churaha_name+":"+churaha_code;
		//values.add(String.valueOf(lat));
		//values.add(String.valueOf(lng));
		values.add(final_data);
		return values;
	}
	
}
